cpp/include/cuvs/neighbors/ivf_flat_helpers.hpp

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright (c) 2023-2024, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
```

Comment:
Also, we should keep the namespace isolation but move these definitions into `ivf_flat.hpp`. Now that cuVS is strictly compiled, there's zero cost for combining these.